### ISSUE TITLE
Focused element deleted by morph

### ISSUE THREAD

**Issue by pdf** [bigskysoftware/idiomorph#55]
I'm using htmx and am looking for more granular DOM updates. After installing the htmx extension from this project and enabling it via `hx-swap="morph"` on the relevant element, the morph occurs however the currently focused element is completely deleted from the DOM.

This happens regularly when a child input element value change triggers a swap of the parent, since the input will always be focused during the swap.

If I specify `hx-swap="morph:{ignoreActive:true}"` this behaviour is avoided, but that seems like a hack, and (IIUC) will ignore changes to the active element that sent by the server, which is undesirable in this case.

**Comment by botandrose** [comment]
@pdf Yes, this is an unfortunate limitation of the browser APIs that we're using to perform the morph. I have three bits of good news, though:
1. Short term: The upcoming v0.5.0 release of Idiomorph has a new `restoreFocus` option, which will restore the focused element if it has an "id" element and focus was lost during the morph, with the caveat of an extra blur and focus event being triggered.
2. Medium term: During the next development cycle I plan to vastly improve the focus retention situation by morphing around the focused element, when possible.
3. Long term: When `moveBefore` https://caniuse.com/mdn-api_document_movebefore lands in browsers, this should become a non-issue, and we already added  support for `moveBefore` in the last v0.4.0 release.

**Comment by botandrose** [comment]
@pdf Should be resolved by the latest v0.7.1 release. Reopen if not!
